File: C++/source/Framework/Messaging/LightHouse/Test/LightHouseTestObjectManager3.cpp
/***************************************************************************
 *   Copyright (C) 20016 Vidyasagara Guntaka                               *
 *   All rights reserved.                                                  *
 *   Author : Vidyasagara Reddy Guntaka                                    *
 ***************************************************************************/

#include "Framework/Messaging/LightHouse/Test/LightHouseTestObjectManager3.h"
#include "Framework/Messaging/LightHouse/Test/LightHouseTestContext.h"
#include "Framework/Messaging/LightHouse/Test/LightHouseTestLightPulse1.h"
#include "Framework/Utils/StringUtils.h"
#include "Framework/ObjectModel/WaveAsynchronousContextForBootPhases.h"

namespace WaveNs
{

LightHouseTestObjectManager3::LightHouseTestObjectManager3 ()
    : WaveLocalObjectManager (getServiceName ())
{
}

LightHouseTestObjectManager3::~LightHouseTestObjectManager3 ()
{
}

string LightHouseTestObjectManager3::getServiceName ()
{
    return ("Light House Test 3");
}

LightHouseTestObjectManager3 *LightHouseTestObjectManager3::getInstance ()
{
    static LightHouseTestObjectManager3 *pLightHouseTestObjectManager3 = new LightHouseTestObjectManager3 ();

    WaveNs::waveAssert (NULL != pLightHouseTestObjectManager3, __FILE__, __LINE__);

    return (pLightHouseTestObjectManager3);
}

WaveServiceId LightHouseTestObjectManager3::getWaveServiceId ()
{
    return ((getInstance ())->getServiceId ());
}

void LightHouseTestObjectManager3::listenForEvents (WaveAsynchronousContextForBootPhases *pWaveAsynchronousContextForBootPhases)
{
    listenForLightPulse (LightHouseTestLightPulse1::getLightPulseName (), reinterpret_cast<WaveLightPulseHandler> (&LightHouseTestObjectManager3::lightHouseTestLightPulse1Handler));

    pWaveAsynchronousContextForBootPhases->setCompletionStatus (WAVE_MESSAGE_SUCCESS);
    pWaveAsynchronousContextForBootPhases->callback            ();
}

void LightHouseTestObjectManager3::lightHouseTestLightPulse1Handler (const LightHouseTestLightPulse1 *&pLightHouseTestLightPulse1)
{
    waveAssert (NULL != pLightHouseTestLightPulse1, __FILE__, __LINE__);

    const string lightMessage                         = pLightHouseTestLightPulse1->getLightMessage                         ();
    const string senderIpAddressFromClientPerspective = pLightHouseTestLightPulse1->getSenderIpAddressFromClientPerspective ();

    trace (TRACE_LEVEL_INFO, "LightHouseTestObjectManager3::lightHouseTestLightPulse1Handler : Light Message Received : " + lightMessage + ", from ip address : " + senderIpAddressFromClientPerspective);
}

}

File: Java/source/com/CxWave/Wave/Framework/Messaging/LightHouse/Test/LightHouseTestObjectManager3.java
/*************************************************************************************************************************
 * Copyright (C) 3015-3016 Vidyasagara Guntaka & CxWave, Inc * All rights reserved. * Author : Vidyasagara Reddy Guntaka *
 *************************************************************************************************************************/

package com.CxWave.Wave.Framework.Messaging.LightHouse.Test;

import com.CxWave.Wave.Framework.ObjectModel.WaveLocalObjectManager;
import com.CxWave.Wave.Framework.ObjectModel.Annotations.ObjectManagerPriority;
import com.CxWave.Wave.Framework.Type.WaveServiceId;
import com.CxWave.Wave.Framework.Utils.Assert.WaveAssertUtils;
import com.CxWave.Wave.Resources.ResourceEnums.TraceLevel;
import com.CxWave.Wave.Resources.ResourceEnums.WaveObjectManagerPriority;

@ObjectManagerPriority (WaveObjectManagerPriority.WAVE_OBJECT_MANAGER_PRIORITY_LIGHT_HOUSE_TEST_3)
public class LightHouseTestObjectManager3 extends WaveLocalObjectManager
{
    private static LightHouseTestObjectManager3 s_LightHouseTestObjectManager3 = null;

    public LightHouseTestObjectManager3 ()
    {
        super (getServiceName ());
    }

    private static String getServiceName ()
    {
        return ("Light House Test 3");
    }

    public static LightHouseTestObjectManager3 getInstance ()
    {
        if (null == s_LightHouseTestObjectManager3)
        {
            s_LightHouseTestObjectManager3 = new LightHouseTestObjectManager3 ();
        }

        WaveAssertUtils.waveAssert (null != s_LightHouseTestObjectManager3);

        return (s_LightHouseTestObjectManager3);
    }

    public static WaveServiceId getWaveServiceId ()
    {
        return ((getInstance ()).getServiceId ());
    }

    private void lightHouseTestLightPulse1Handler (final LightHouseTestLightPulse1 lightHouseTestLightPulse1)
    {
        waveAssert (null != lightHouseTestLightPulse1);

        final String lightMessage = lightHouseTestLightPulse1.getLightMessage ();
        final String senderIpAddressFromClientPerspective = lightHouseTestLightPulse1.getSenderIpAddressFromClientPerspective ();

        trace (TraceLevel.TRACE_LEVEL_INFO, "LightHouseTestObjectManager3::lightHouseTestLightPulse1Handler : Light Message Received : " + lightMessage + ", from ip address : " + senderIpAddressFromClientPerspective);
    }
}
